wom/src/test/scala/wom/types/WomMapTypeSpec.scala

```diff
@@ -57,14 +57,47 @@ class WomMapTypeSpec extends FlatSpec with Matchers  {
       case Failure(_) => fail("Failed to coerce a map to an object")
     }
   }
-  it should "detect invalid map construction if there are mixed types" in {
+
+  it should "detect invalid map construction if there are mixed keys" in {
     try {
-      WomMap(WomMapType(WomStringType, WomStringType), Map(WomInteger(0) -> WomString("foo"), WomString("x") -> WomInteger(2)))
+      WomMap(WomMapType(WomStringType, WomStringType), Map(WomInteger(0) -> WomString("foo"), WomString("x") -> WomString("y")))
       fail("Map initialization should have failed")
     } catch {
-      case _: UnsupportedOperationException => // expected
+      case e: UnsupportedOperationException =>
+        e.getMessage shouldEqual "Cannot construct WomMapType(WomStringType,WomStringType) with mixed types in map keys: [WomInteger(0), WomString(x)]"
     }
   }
+
+  it should "detect invalid map construction if there are mixed values" in {
+    try {
+      WomMap(WomMapType(WomStringType, WomStringType), Map(WomString("bar") -> WomString("foo"), WomString("x") -> WomInteger(2)))
+      fail("Map initialization should have failed")
+    } catch {
+      case e: UnsupportedOperationException =>
+        e.getMessage shouldEqual "Cannot construct WomMapType(WomStringType,WomStringType) with mixed types in map values: [WomString(foo), WomInteger(2)]"
```

Comment:
This definitely *feels* like a bug...  I would hope that we'd have no problem doing that coercion from Int to String

Reply:
Very possible, but feels out of scope for this PR

Reply:
We should discuss this one

Reply:
I do think we ought to this an issue with this exact example as the error case? I'd add labels `Bug`/`Redwood`.

FWIW I wanted to also suggest `Comply with language spec` but the spec is _probably_ silent on the exact point of what is acceptable if the types are mixed in `Map` construction but would be coercible, it might be more a case of "Update language spec comply with what is intuitively right"